stage0/src/zero_page.rs

```diff
@@ -59,6 +79,39 @@ pub fn init_zero_page(fw_cfg: &mut FwCfg) -> &'static mut BootParams {
     zero_page
 }
 
+/// Builds an E820 table by reading the low and high memory amount from CMOS.
+///
+/// The code is largely based on what SeaBIOS is doing, but
```

Comment:
If you have permalinks to the SeaBIOS code that inspired this, it would also be helpful for future reference.

Reply:
Done.